third_party/nvfuser/csrc/executor.cpp

```diff
@@ -1111,11 +1109,67 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
     launch_params_ =
         computeLaunchParams(launch_constraints, expr_eval, warp_size_);
 
+    auto alias_indices_entry =
+        executor_utils::caching::ExecutorCompileTimeEntry<
+            executor_utils::caching::InputAliasIndices>(
+            compileTimeDataCache(), [&]() {
+              return std::make_unique<std::vector<std::pair<int, int>>>(
+                  fusion_->getInputAliasIndices());
+            });
+
+    auto& alias_indices = alias_indices_entry.get();
+
+    // NOLINTNEXTLINE(bugprone-branch-clone)
+    if (outputs.empty()) {
```

Comment:
Can you mention here that outputs have to be added to the kernel argument holder before checking if recompilation is required? We should clean up this function as it's way too long with deep branches (not your fault), but some comments like this would be helpful to understand the context.

Reply:
Done